tests/unit/test_charm.py

```diff
@@ -62,22 +55,22 @@ def test_pebble_ready(self):
             BlockedStatus("Pebble ready, waiting for Spark Configuration"),
         )
 
-    def test_config_changed(self):
-        with patch("ops.model.Container.list_files", return_value=None):
-            # Ensure the simulated Pebble API is reachable
-            self.harness.set_can_connect(CONTAINER, True)
-            # Trigger a config-changed event with an updated value
-            self.harness.update_config({CONFIG_KEY_S3_ENDPOINT: "http://S3_SERVER:S3_PORT"})
-            self.harness.update_config({CONFIG_KEY_S3_ACCESS_KEY: "S3_ACCESS_KEY"})
-            self.harness.update_config({CONFIG_KEY_S3_SECRET_KEY: "S3_SECRET_KEY"})
-            self.harness.update_config({CONFIG_KEY_S3_LOGS_DIR: "S3_LOGS_DIR"})
-            self.harness.update_config(
-                {
-                    CONFIG_KEY_S3_CREDS_PROVIDER: "org.apache.hadoop.fs.s3a.SimpleAWSCredentialsProvider"
-                }
-            )
-            self.harness.update_config({CONFIG_KEY_S3_SSL_ENABLED: False})
-
-            self.assertEqual(
-                self.harness.model.unit.status, ActiveStatus("Spark log directory: S3_LOGS_DIR")
-            )
+    # def test_config_changed(self):
```

Comment:
Are these commented lines done on purpose?  Here you can unit test your charm when it is relating with the S3-integrator-charm. You can take inspiration from this [PR](https://github.com/canonical/mongodb-operator/pull/156/files)